### PR TITLE
Metadata update

### PR DESCRIPTION
**Pull request by rinigus**
Here, handling of the packages is revised as follows:

- Chum class keeps a container with all packages
- Packages info is updated via Chum only
- On repository package list update, all metadata from SPEC is acquired as well, including extra Chum SPEC data from `%description` as well as installed version of the package

Current state is WIP, as new system requires integration with GUI (package listing and package display). This integration is coming up in the next commits to this branch. However, it allows you to look into the changes and pull the breaks early if needed.

I plan to continue with adjusting GUI to get data from that package container in Chum through some proxy list classes. Those are expected to replace the current packages models.

### CODE REVIEW

**Comment by mentaljam** [comment]
Hi!

I'm just starting to work on list queries. I'll pull this commits to my branch to work on the top of it

**Comment by rinigus** [comment]
This is ready for review. It is not fully hooked to GUI in terms of package page, but the main implementation is in place. In general, model for packages list is ChumPackagesModel. Through its properties, it will be possible to filter the lists. Right now, it is shown how to use it for all packages and only the updated ones. While search property is there, it is not hooked to search yet.

PackagesListPage is old PackagesPage and it is able to replace Update and AvailablePackages, unless I dropped some functionality by mistake. I hope it is easy to follow new code and it makes sense.

Please review when you get time and if it is OK, merge to `metadata` branch. 

@mentaljam: thanks - looking forward!

I will work on parsing YAML entries to proper properties in ChumPackage C++ class.

**Comment by rinigus** [comment]
Metadata is now parsed in C++ into ChumPackage properties. The standard has been updated a bit, @piggz already merged the changes. @mentaljam, maybe you can rebase your branch or we can merge it all together later

**Comment by rinigus** [comment]
I have added search support, but not very happy about it. Currently, you have to press enter to activate it. This is due to the loss of focus if I let it search on every text change leading to dropping of the keyboard. So, it is a compromise - at least it should work. 

Maybe one day someone will figure out how to work around focus issue. But there are probably more important issues to fix

**Comment by rinigus** [comment]
Data loading from GitHub has been added. I had to move token out of the source code as Github immediately disabled it as soon as it was found in a commit. You can grab token file from OBS (in :testing right now, later will move to the main one).

**Comment by rinigus** [comment]
In 162cb0f29d17d888fa07b26e1e8f22f51aec7726, listing of installed packages is fixed. In this respect, we don't need zypper support anymore.

**Comment by mentaljam** [comment]
Hi!

Sorry for disappearing. I've spent the previous Sunday attempting to configure and run SDK with no success. Yesterday I started with a clean installation but again SDK just doesn't work because of some SSH errors. I haven't found a quick solution. Also I started a review but this PR now contains so much commits that I don't feel myself able to look through all of them and understand them. Sorry to let you down but I think you should continue the current work without my participation

**Comment by rinigus** [comment]
> Also I started a review a week ago but this PR contains now so much commits that I don't feel myself able to look through all of them and understand them. Sorry to let you down but I think you should continue the current work without my participation

That's understandable. I just moved on as it seemed like pieces were falling into places. Hopefully, you can help us when you get SDK working and time for it. I'll look into your review above, thank you!

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,13 @@
+add_compile_options(
```
[the remark this answers is not in the document]
Comment (rinigus) [reply src/CMakeLists.txt]: Thanks, changed accordingly

File: src/chumpackage.cpp [hunk 2]
```diff
@@ -12,62 +13,88 @@ ChumPackage::ChumPackage(QObject *parent)
...
+    if (m_name.startsWith(prefix))
+      m_name = m_name.mid(prefix.length());
+  QStringList nparts = m_name.split('-');
+  m_name = QString{};
+  for (const QString& b: nparts)
+    m_name += b.left(1).toUpper() + b.mid(1).toLower() + " ";
+  m_name = m_name.trimmed();
+
+  // parse description
+  QStringList descLines = m_description.split(QRegularExpression("(?m)^\\s*$"), QString::SkipEmptyParts);
```
[the remark this answers is not in the document]
Comment (rinigus) [reply src/chumpackage.cpp:103]: I find it easier to read as it is now. There are other operations done on those sections (we don't know how many we have in each RPM SPEC) and it makes sense to split it right at the front. So, I would suggest to keep as it is.

File: src/chumpackage.cpp [hunk 3]
```diff
@@ -12,62 +13,88 @@ ChumPackage::ChumPackage(QObject *parent)
...
+  m_size        = v.size();
+
+  // derive name
+  m_name = Daemon::packageName(m_pkid);
+  for (QString prefix: {"harbour-", "openrepos-"})
+    if (m_name.startsWith(prefix))
+      m_name = m_name.mid(prefix.length());
+  QStringList nparts = m_name.split('-');
+  m_name = QString{};
+  for (const QString& b: nparts)
```
[the remark this answers is not in the document]
Comment (rinigus) [reply src/chumpackage.cpp]: Thanks, your version is used now

File: src/chumpackage.cpp [hunk 4]
```diff
@@ -12,62 +13,88 @@ ChumPackage::ChumPackage(QObject *parent)
...
+
+  m_description = v.description();
+  m_summary     = v.summary();
+  m_url         = v.url();
+  m_license     = v.license();
+  m_size        = v.size();
+
+  // derive name
+  m_name = Daemon::packageName(m_pkid);
+  for (QString prefix: {"harbour-", "openrepos-"})
```
[the remark this answers is not in the document]
Comment (rinigus) [reply src/chumpackage.cpp]: QLatin1String is used now

**Comment by rinigus** [comment]
Merging as suggested by @piggz

**Pull request merged**